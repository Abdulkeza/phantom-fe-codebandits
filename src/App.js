import React, { useState, useEffect } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import './app.css';
import Footer from './components/Footer.js';
import Header from './components/Header.js';
import SkeletonScreen from './components/SkeletonScreen.js';
import Account from './pages/Account.js';
import LandingPage from './pages/LandingPage.js';
import Login from './pages/Login.js';
import { store } from './redux/store.js';

function App() {
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    setLoading(true);
    const timer = setTimeout(() => {
      setLoading(false);
    }, 3000);

    return () => clearTimeout(timer);
  }, []);

  return (
    <div className="App">
      {loading && <SkeletonScreen />}
      {!loading && (
        <div>
          <Provider store={store}>
            <BrowserRouter>
              <Header />
              <Routes>
                <Route path="/" element={<LandingPage />} />
                <Route path="/login" element={<Login />} />
                <Route path="/account" element={<Account />} />
              </Routes>
              <Footer />
            </BrowserRouter>
          </Provider>
        </div>
      )}
    </div>
  );
}

export default App;
