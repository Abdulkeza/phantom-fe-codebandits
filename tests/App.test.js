import React from "react";
import { shallow } from "enzyme";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import App from "../src/App";
import { counterReducer } from "../src/redux/reducers/counterReducer";
import reducer from '../src/redux/store'
import { increment } from "../src/redux/reducers/counterReducer";

// import App from '.';



const mockStore = configureStore([]);

describe("Test App Entry Point", () => {
    
  it("Should have a header tag with Phantom", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find("h1").text()).toEqual("Phantom");
  });

  it("Should have a p tag with powered by CodeBandits", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find("p").text()).toEqual("powered by CodeBandits");
  });
});



describe('COunter component', ()=>{
  it('renders the count', ()=>{

    const action = {
      dispatch: jest.fn(),
      value: 3,
    }

    const wrapper = shallow(<App {...action} />)
    expect(wrapper.find('h3')).toEqual(3)
  })

  it('dispatches the right action for incrementing', ()=>{

    const action = {
      dispatch: jest.fn(),
      value: 3,
    }

    const wrapper = shallow(<App {...action} />);
    wrapper.find('button.increment').simulate('click');

    expect(action.dispatch).toHaveBeenCalledWith(counterReducer(action.value))
  })

  it('dispatches the right action for decrementing', ()=>{

    const action = {
      dispatch: jest.fn(),
      value: 3,
    }

    const wrapper = shallow(<App {...action} />);
    wrapper.find('button.decrement').simulate('click');

    expect(action.dispatch).toHaveBeenCalledWith(counterReducer(action.value))
  })

})




// describe('My Connected React-Redux Component', () => {
//   let store;
//   let component;

//   beforeEach(() => {
//     store = mockStore({
//       myState: 'sample text',
//     });

//     component = renderer.create(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );
//   });

//   it('should render with given state from Redux store', () => {
//     expect(component.toJSON()).toMatchSnapshot();
//   });

//   it('should dispatch an action on button click', () => {

//   });
// });


