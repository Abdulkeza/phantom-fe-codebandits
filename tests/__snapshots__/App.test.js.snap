// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should render App 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="flex bg-background border-b-4 pb-4 sticky top-0 z-20"
    >
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Phantom"
            className="w-44 xl:w-60 ml-8 xl:ml-20 mt-7 mb-2"
            src="phantom.png"
          />
        </a>
      </div>
      <div
        className="hidden md:flex xl:flex ml-auto mr-20"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          <div>
            <button
              className="bg-primary hover:bg-hover text-white rounded-xl mt-7 ml-6 rounded-xl py-3 px-5"
              type="button"
            >
              For operators
            </button>
          </div>
        </a>
        <div>
          <button
            className="border border-primary bg-background hover:bg-hover2 text-primary rounded-xl mt-7 ml-6 px-6 rounded-xl py-3 px-5"
            type="button"
          >
            Track Bus
          </button>
        </div>
        <div>
          <button
            className="undefined md:mt-8 lg:mt-8 xl:mt-8 flex py-2 px-4 text-lg hover:font-bold"
            onClick={[Function]}
            type="button"
          >
            <h1>
              English
            </h1>
            <img
              alt="dropdown"
              className="w-3 mt-3 ml-2 h-2"
              src="dropdown.png"
            />
          </button>
        </div>
      </div>
      <section
        className="flex flex-col absolute right-0 bg-background px-2"
      >
        <button
          className="md:hidden xl:hidden ml-auto mr-12 mt-8"
          id="toggle"
          onClick={[Function]}
          type="button"
        >
          <div>
            <div
              className="h-1 w-7 bg-primary mt-1"
            />
            <div
              className="h-1 w-7 bg-primary mt-1"
            />
            <div
              className="h-1 w-7 bg-primary mt-1"
            />
          </div>
        </button>
      </section>
    </div>
    <div>
      <div
        className="flex flex-col items-center lg:flex-row xl:flex-row"
      >
        <div>
          <h1
            className="flex flex-col text-3xl md:text-5xl xl:text-5xl font-raleway font-extrabold mx-12 xl:mx-28 mt-24 md:mt-32 xl:mt-32"
          >
            <span>
              Track bus location,
            </span>
            <span
              className="ml-4 md:ml-10"
            >
              Save your time.
            </span>
          </h1>
          <p
            className="flex flex-col text-sm md:text-xl xl:text-base mx-20 md:ml-32 xl:ml-52 mt-10"
          >
            <span>
              Check the location of the buses
            </span>
            <span
              className="ml-4"
            >
              on their respective routes
            </span>
          </p>
          <div>
            <button
              className="bg-primary hover:bg-hover text-white rounded-3xl mt-12 px-12 xl:px-12 py-2 ml-28 md:ml-48 xl:ml-64 rounded-xl py-3 px-5"
              type="button"
            >
              Start
            </button>
          </div>
        </div>
        <div
          className="w-full px-20 lg:w-80 xl:w-96 lg:ml-auto xl:ml-auto lg:mr-36 xl:mr-36 lg:px-0 xl:px-0 mt-28"
        >
          <img
            alt="Bus"
            src="bus.png"
          />
        </div>
      </div>
      <div
        className="flex flex-col items-center"
      >
        <h1
          className="text-4xl font-raleway font-extrabold mt-28 xl:mt-44 mb-10 text-center"
        >
          How this Works
        </h1>
        <div
          className=" flex flex-col lg:flex-row xl:flex-row mt-12 mb-32"
        >
          <div
            className="xl:justify-start xl:pt-2 border w-52 h-52 px-4 text-center flex flex-col justify-center items-center rounded-lg shadow-md"
          >
            <img
              alt="start"
              className="w-12 mt-4"
              src="start.png"
            />
            <h3
              className="font-raley mt-2 font-extrabold text-sm"
            >
              1. Start
            </h3>
            <p
              className="font-sans text-center text-sm my-2"
            >
              Click the starting button, to begin tracking buses
            </p>
          </div>
          <div
            className="my-28 lg:my-0 xl:my-0 lg:mx-32 xl:mx-32 border w-52 h-52 px-4 text-center flex flex-col justify-center items-center rounded-lg shadow-md"
          >
            <img
              alt="locator"
              className="w-12 mt-4"
              src="locator.png"
            />
            <h3
              className="font-raley mt-2 font-extrabold text-sm"
            >
              2. Enter location & destination
            </h3>
            <p
              className="font-sans text-center text-sm my-2"
            >
              In the respective fields, enter your current location and destination
            </p>
          </div>
          <div
            className="undefined border w-52 h-52 px-4 text-center flex flex-col justify-center items-center rounded-lg shadow-md"
          >
            <img
              alt="globe"
              className="w-12 mt-4"
              src="globe.svg"
            />
            <h3
              className="font-raley mt-2 font-extrabold text-sm"
            >
              1. View the location of the nearest bus
            </h3>
            <p
              className="font-sans text-center text-sm my-2"
            >
              Through the map, you will see the location of the buses you want to track
            </p>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="flex bg-primary xl:justify-center md:text-xl text-background font-sans font-sm py-8 lg:py-2 xl:py-2 mt-auto"
    >
      <div
        className="flex flex-col lg:flex-row xl:flex-row px-12 text-center items-center"
      >
        <img
          alt="Codebandits"
          className="w-24 ml-2 lg:ml-8 lg:mt-4 xl:mt-2"
          src="codebandits.png"
        />
        <h3
          className="lg:ml-8 xl:ml-20 mt-6 xl:mt-2"
        >
          Â©copyright Codebandits 2022
        </h3>
      </div>
      <div
        className="flex flex-col md:text-xl lg:flex-row xl:flex-row lg:mt-7 xl:mt-3 mr-16 md:mx-20 xl:ml-20 ml-auto md:ml-auto"
      >
        <a
          className="xl:ml-0 lg:ml-0 mt-2 text-md mb-4 hover:font-bold"
          href="/"
          onClick={[Function]}
        >
          Support
        </a>
        <a
          className="xl:ml-28 lg:ml-12 mt-2 text-md mb-4 hover:font-bold"
          href="/"
          onClick={[Function]}
        >
          Terms of use
        </a>
        <a
          className="xl:ml-28 lg:ml-12 mt-2 text-md hover:font-bold"
          href="/login"
          onClick={[Function]}
        >
          Administration
        </a>
      </div>
    </footer>
  </div>
</div>
`;
